website/src/content/main-concepts/proxy-to-external-api.mdx

```diff
@@ -16,74 +18,213 @@ Merchant Center applications can make [authenticated requests](/main-concepts/au
 
 Since your backend server is hosted on a different domain, you can't send authenticated requests to your server from your Custom Application. Well, you can but you don't know if the request comes from an authenticated user or not.
 
-# Validating authenticated requests
+# Proxying requests to the external API
+
+To be able to validate that the request comes from an authenticated user, the Merchant Center API Gateway provides an endpoint that must be used to connect to your external API in a secured manner.
+
+```
+/proxy/forward-to
+```
+
+Requests to that endpoint should additionally pass the following HTTP headers:
 
-To be able to validate that the request comes from an authenticated user, we implemented a mechanism to allow your server to verify the validity of the `mcAccessToken`.
+- `Accept-version: v2`: See [Versioning](#versioning).
+- `X-Forward-To: <url>`: The URL to your external API.
+- `X-Project-Key: <project-key>`: The key of the project currently being used by the Custom Application. The Merchant Center API Gateway will perform a validation check to ensure that the user has access to the project, then forward the request to your server only if the check was successful.
 
-Requests to your backend server should be configured as following:
+To facilitate the usage of the [built-in Apollo client and the SDK actions](/main-concepts/data-fetching), we provide some utility helpers to abstract away the request configuration for the HTTP headers listed above.
 
-- send the request to `/proxy/forward-to` endpoint of the Merchant Center API Gateway.
-- define the original request URL to your backend server as an HTTP header `X-Forward-To`.
+## Usage for Apollo
+
+We can leverage the `context` option for Apollo queries to change a bit how the request is configured. The `@commercetools-frontend/application-shell` package now exposes an utility function to configure the Apollo context for the `/proxy/forward-to` usage.
 
 ```js
-fetch(`https://mc-api.commercetools.com/proxy/forward-to`, {
-  method: 'GET',
-  headers: {
-    Accept: 'application/json',
-    'X-Forward-To': 'https://my-custom-app.com/foo/bar',
-  },
-  credentials: 'include',
+import { useQuery } from 'react-apollo';
+import { createApolloContextForProxyForwardTo } from '@commercetools-frontend/application-shell';
+const Fetcher = () => {
+  const { loading, data, error } = useQuery(MyQuery, {
+    context: createApolloContextForProxyForwardTo({
```

Comment:
This is technically correct but I would suggest to set it up like this.

I would suggest that people use a `forward-to-apollo-contexts.js`-module

```js
const customServiceApolloContext = createApolloContextForProxyForwardTo({
  uri: 'https://my-custom-app.com/graphql',
})

export { customServiceApolloContext } 
```

to then more convienently do:

```js
import { customServiceApolloContext } from './forward-to-apollo-contexts'

const { loading, data, error } = useQuery(MyQuery, {
   context: customServiceApolloContext
})
```

cause I would assume they have many queries and mutations.

Reply:
It's not that simple. Most likely the URL is defined in the `env.json`, so it's passed through the app context.

I could maybe update the example with that, if that makes it clearer.

Defining the config statically only works if you hardcode the URL.

Reply:
Yeah makes sense. I wonder though how many have variable endpoints. I'd make the docs work for the majority center best practices around that and no have minior edge cases influence the majority.

Reply:
If we plan support variable endpoints then we should maybe have a `useProxyForwardToContext`-hook which people can use and which encapsulates a respective or configurable env.json value

```js
const forwardToCustomServiceContext = useProxyForwardToContext({ envKey: 'foo' })
const { loading, data, error } = useQuery(MyQuery, {
   context: forwardToCustomServiceContext
})
```

Reply:
In general you would always want to use variables. I also saw a couple of cases where `env.json` has been used. Which is actually how it was intended to be used.

I understand your point of encapsulating things but it just gets hard to find the balance to accommodate different use cases.

I think I would just update the doc example to read the value from the app context. People can then decide on their own how to access the URL.

Reply:
Makes sense. Thanks for taking things into consideration!